specifyweb/specify/load_datamodel.py

```diff
@@ -307,6 +307,9 @@ def flag_system_tables(datamodel: Datamodel) -> None:
     'Commonnametx.citations',
     'Conservdescription.events',
     'Deaccession.deaccessionagents',
+    'Deaccession.disposals',
+    'Deaccession.gifts',
+    'Deaccession.exchangeouts',
```

Comment:
Adding disposals, gifts and exchangeouts to dependent fields so that we can just add up totalPreps and totalItems from those objects. Let me know if there's a better way to do this

Reply:
ya, that's a good way to do it. 👍

Reply:
Hmm without further research, this can be a bad change. For example, now in wb, gifts can not be shared between deaccessions. Check if we have business logic that already prevents that. If so, then this is fine!

Reply:
Also, dependents get deleted when primary gets deleted. Do we also have a business rule for that? Will this change cause that?

Reply:
This is not the way to fix it as these are not dependent on the Deaccessions and has unintended consequences as mentioned by @realVinayak.

Reply:
That's fine, we can just do this instead in the `calculate_extra_fields` code for Deaccession:
```python
Disposal = get_model("Disposal")
Exchangeout = get_model("ExchangeOut")
Gifts = get_model("Gift")
disposals = Disposal.objects.filter(deaccession=obj)
gifts = Gifts.objects.filter(deaccession=obj)
exchangeouts = Exchangeout.objects.filter(deaccession=obj)
```

Reply:
👑 @acwhite211 @sharadsw